spatialmath/mesh.go

```diff
@@ -539,3 +544,56 @@ func calculatePolygonAreaWithTriangulation(vertices []r3.Vector) float64 {
 		return totalArea
 	}
 }
+
+// trianglesToPLYBytes converts the mesh's triangles to bytes in PLY format.
+func (m *Mesh) trianglesToPLYBytes() []byte {
+	// Collect all unique vertices and create vertex-to-index mapping
+	vertexMap := make(map[string]int)
+	vertices := make([]r3.Vector, 0)
+
+	for _, tri := range m.triangles {
+		for _, pt := range tri.Points() {
+			scaledPt := r3.Vector{X: pt.X / 1000.0, Y: pt.Y / 1000.0, Z: pt.Z / 1000.0}
+			key := fmt.Sprintf("%.10f,%.10f,%.10f", scaledPt.X, scaledPt.Y, scaledPt.Z)
+			if _, exists := vertexMap[key]; !exists {
+				vertexMap[key] = len(vertices)
+				vertices = append(vertices, scaledPt)
+			}
+		}
+	}
+
+	var buf bytes.Buffer
+
+	// Write PLY header
+	buf.WriteString("ply\n")
+	buf.WriteString("format ascii 1.0\n")
+	buf.WriteString(fmt.Sprintf("element vertex %d\n", len(vertices)))
+	buf.WriteString("property float x\n")
+	buf.WriteString("property float y\n")
+	buf.WriteString("property float z\n")
+	buf.WriteString(fmt.Sprintf("element face %d\n", len(m.triangles)))
+	buf.WriteString("property list uchar int vertex_indices\n")
+	buf.WriteString("end_header\n")
+
+	// Write vertices
+	for _, vertex := range vertices {
+		buf.WriteString(fmt.Sprintf("%f %f %f\n", vertex.X, vertex.Y, vertex.Z))
```

Comment:
same file as above, this is what I have for this section
```
-0.11941 0.0875899 0.332115 0.094139 0.521399 -0.848104
-0.127854 0.0857999 0.32863 0.103633 0.542683 -0.83352
-0.0670865 0.0770517 0.330627 0.106932 0.550113 -0.828216
```

Reply:
Sure, but if we don't transmit the normals, there will only be 3 values